bson/vector.go

```diff
@@ -0,0 +1,266 @@
+// Copyright (C) MongoDB, Inc. 2024-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package bson
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math"
+)
+
+// These constants are vector data types.
```

Comment:
Optional: This comment actually gets rendered in the docs. Consider a more descriptive comment.

E.g.
```go
// BSON binary vector types as described in https://bsonspec.org/spec.html.
```